packages/fast-tooling-react/src/data-utilities/generate.spec.ts

```diff
@@ -0,0 +1,345 @@
+import "jest";
+import { getDataFromSchema } from "./generate";
+import { ChildOptionItem } from ".";
+
+/**
+ * Gets an example from a schema
+ */
+describe("getDataFromSchema", () => {
+    test("should return an enum value", () => {
+        const schema: any = {
+            enum: ["foo", "bar"],
+        };
+
+        expect(getDataFromSchema(schema)).toEqual("foo");
+    });
+    test("should return a string", () => {
+        const schema: any = {
+            type: "string",
+        };
+
+        expect(typeof getDataFromSchema(schema)).toBe("string");
+
+        const schemaWithDefault: any = {
+            type: "string",
+            default: "foo",
+        };
+
+        expect(typeof getDataFromSchema(schemaWithDefault)).toBe("string");
+        expect(getDataFromSchema(schemaWithDefault)).toEqual("foo");
+    });
+    test("should return a boolean", () => {
+        const schema: any = {
+            type: "boolean",
+        };
+
+        expect(typeof getDataFromSchema(schema)).toBe("boolean");
+
+        const schemaWithDefault: any = {
+            type: "boolean",
+            default: false,
+        };
+
+        expect(typeof getDataFromSchema(schemaWithDefault)).toBe("boolean");
+        expect(getDataFromSchema(schemaWithDefault)).toEqual(false);
+    });
+    test("should return a number", () => {
+        const schema: any = {
+            type: "number",
+        };
+
+        expect(typeof getDataFromSchema(schema)).toBe("number");
+
+        const schemaWithDefault: any = {
+            type: "number",
+            default: 1,
+        };
+
+        expect(typeof getDataFromSchema(schemaWithDefault)).toBe("number");
+        expect(getDataFromSchema(schemaWithDefault)).toEqual(1);
+    });
+    test("should return null", () => {
+        const schema: any = {
+            type: "null",
+        };
+
+        expect(getDataFromSchema(schema)).toBe(null);
+    });
+    test("should return an enum specified value", () => {
+        const schema: any = {
+            enum: ["foo", "bar", "bat"],
+        };
+
+        expect(getDataFromSchema(schema)).toBe("foo");
+    });
+    test("should return default even if enums are specified", () => {
```

Comment:
What happens if "default" is not in the enum?

Reply:
Theoretically then the default value would not validate against the schema, which is a little ambiguous in the way since it "should" valudate. It is indicated that the default value should be used to generate a key/value pair per https://json-schema.org/understanding-json-schema/reference/generic.html